libraries/rush-lib/src/logic/operations/AsyncOperationQueue.ts

```diff
@@ -5,7 +5,17 @@ import { OperationExecutionRecord } from './OperationExecutionRecord';
 import { OperationStatus } from './OperationStatus';
 
 /**
- * Implmentation of the async iteration protocol for a collection of IOperation objects.
+ * When the queue returns an unassigned operation, it means there is no workable operation at the time,
+ * and the caller has a chance to make a decision synchronously or asynchronously:
+ * 1. Manually invoke `tryGetRemoteExecutingOperation()` to get a remote executing operation.
+ * 2. Or, return in callback or continue the for-loop, which internally invoke `assignOperations()` to assign new operations.
```

Comment:
In either option 1 or option 2, if there is no remote executing operation available, this will create an async tick-long loop and burn CPU cycles until new work is available, rather than simply waiting until the next operation is available.

Reply:
That's true. The caller should carefully make a decision. I've not found a clear way to archive "simple waiting until the next operation is available". Any thoughts are welcome.